Jint/IConstraint.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Jint
+{
+    public interface IConstraint
```

Comment:
could probably be internal, otherwise might need some documentation if intended for users to extend/specialize

Reply:
I wanted to make a constraint that allows me to pause the time. The goal is to use scripting for async calls and only measure the time that is actually spend within the script (CPU bound) not for "waiting".

Furthermore the memory constraint needs to be enhanced to support async calls and potential thread switches.

Reply:
One thing that came to mind also is that if this were abstract base class, dispatch would be a bit cheaper, all invocations are in hot path.

Reply:
I have profiled it because of the DateTime.UtcNow and I think this is a micro-micro-micro-optimization with very little gain, probably the same like my for loop :D